packages/flutter_test/test/widget_tester_test.dart

```diff
@@ -699,6 +701,18 @@ void main() {
       ticker.stop();
   });
 
+  group('testWidgets matrix works', () {
+    int numberOfVariationsRun = 0;
+
+    testWidgets('matrix tests run all variations', (WidgetTester tester) async {
+      if (debugDefaultTargetPlatformOverride == null) {
+        expect(numberOfVariationsRun, equals(TargetPlatform.values.length));
+      } else {
+        numberOfVariationsRun += 1;
+      }
+    }, matrix: <TestDimension<dynamic>>[ TargetPlatformDimension(TargetPlatform.values), DefaultDimension() ]);
+  });
+
```

Comment:
we'll need more tests, there's a lot more behaviour here than this test verifies.